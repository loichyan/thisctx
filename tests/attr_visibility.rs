use thisctx::IntoError;

mod error {
    use thisctx::WithContext;

    #[derive(Debug, WithContext)]
    #[thisctx(visibility = "pub(crate)")]
    // Syntax as follows is also allowed.
    // #[thisctx(visibility(pub(crate)))]
    // #[thisctx(pub(crate))]
    pub enum Error {
        #[thisctx(visibility = "pub")]
        PubVariant(#[thisctx(visibility = "pub(crate)")] i32),
        PubCrateVariant(i32),
    }
}

#[test]
fn attr_visibility() {
    error::PubVariantContext(0).build();
    error::PubCrateVariantContext(0).build();
}
