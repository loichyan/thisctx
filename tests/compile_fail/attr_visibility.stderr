error[E0364]: `PubCrateVariant` is private, and cannot be re-exported
  --> tests/compile_fail/attr_visibility.rs:19:9
   |
19 | pub use error::PubCrateVariant;
   |         ^^^^^^^^^^^^^^^^^^^^^^
   |
note: consider marking `PubCrateVariant` as `pub` in the imported module
  --> tests/compile_fail/attr_visibility.rs:19:9
   |
19 | pub use error::PubCrateVariant;
   |         ^^^^^^^^^^^^^^^^^^^^^^

error[E0603]: struct `PrivateVariant` is private
  --> tests/compile_fail/attr_visibility.rs:18:16
   |
18 | pub use error::PrivateVariant;
   |                ^^^^^^^^^^^^^^ private struct
   |
note: the struct `PrivateVariant` is defined here
  --> tests/compile_fail/attr_visibility.rs:6:14
   |
6  |     #[derive(WithContext)]
   |              ^^^^^^^^^^^
   = note: this error originates in the derive macro `WithContext` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0616]: field `0` of struct `PrivateField` is private
  --> tests/compile_fail/attr_visibility.rs:28:7
   |
28 |     t.0;
   |       ^ private field
